rclcpp/src/rclcpp/executor.cpp

```diff
@@ -40,30 +40,80 @@ using rclcpp::Executor;
 using rclcpp::ExecutorOptions;
 using rclcpp::FutureReturnCode;
 
+namespace
+{
+
+std::shared_ptr<rcl_guard_condition_t>
+create_shutdown_guard_condition(std::shared_ptr<rclcpp::Context> context)
```

Comment:
In order to maintain the same level of coverage, I think the failure cases in this function will need to be covered with new mock tests.

Reply:
Used `rclcpp::GuardCondition`, which already has 100% test coverage.

Reply:
See https://github.com/ros2/rclcpp/pull/1404/commits/564993c3e2920e4205fae2118d545de34bf07f0f